Add go1.12.2 / go1.11.7

freeformz opened the pull request:
Drop support for `Go.SupportsModuleExperiment`. It's not really
and expermiment anymore and any version of go >= go1.11 will be
supporting it.

Drop support for `Go.Supported` since the buildpack isn't using it
anymore. :-(

Commit:
Add go1.12.2 / go1.11.7

Drop support for `Go.SupportsModuleExperiment`. It's not really
and expermiment anymore and any version of go >= go1.11 will be
supporting it.

Drop support for `Go.Supported` since the buildpack isn't using it
anymore. :-(

jmorrell commented on lib/common.sh:
```diff
@@ -338,16 +338,16 @@ determineTool() {
             warn ""
         fi
-        if ! <"${DataJSON}" jq  -e '.Go.SupportsModuleExperiment | any(. == "'${ver}'")' &> /dev/null; then
+        if ! echo "\"${ver}\"" | jq -e 'split(".") | map(gsub("go";"")) | map(tonumber) | .[0] >= 1 and .[1] >= 11' &> /dev/null; then
```
A comment would be very helpful here. This line is very difficult to parse

```
# if using 1.11 or older
```

`.[0] >= 1 and .[1] >= 11` is particularly confusing because it looks like it's testing whether that is true instead of catching when it fails. Wrapping this into a `goVersionSupportsModules` function might make this easier for others.

freeformz replied:
done.

Commit:
Address feedback

Commit:
Skip the vendored mattes migrate install on cedar:14

gcc throws the following error:

/tmp/output.Pc7s/cache/go1.12.2/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/usr/bin/ld: /tmp/go-link-758220591/000020.o: unrecognized relocation (0x2a) in section `.text'
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
